RestaurantReservation.API/Controllers/OrderItemController.cs

```diff
@@ -0,0 +1,262 @@
+        }
+
+        var orderItems = await _orderItemRepository.GetOrderItemsForOrderAsync(reservationId, orderId);
+        return Ok(_mapper.Map<IEnumerable<OrderItemDto>>(orderItems));
+    }
+
+    /// <summary>
+    /// Gets a specific order item by ID for a specific order in a reservation.
+    /// </summary>
+    /// <param name="reservationId">The ID of the reservation.</param>
+    /// <param name="orderId">The ID of the order.</param>
+    /// <param name="orderItemId">The ID of the order item to retrieve.</param>
+    /// <returns>The order item data transfer object.</returns>
+    /// <response code="200">Returns the requested order item.</response>
+    /// <response code="404">If the reservation, order, or order item with the specified IDs is not found.</response>
+    [HttpGet("{orderItemId}", Name = "GetOrderItem")]
+    [ProducesResponseType(StatusCodes.Status200OK, Type = typeof(OrderItemDto))]
+    [ProducesResponseType(StatusCodes.Status404NotFound)]
+    public async Task<ActionResult<OrderItemDto>> GetOrderItem(int reservationId, int orderId, int orderItemId)
+    {
+        var reservationExists = await _reservationRepository.ReservationExistsAsync(reservationId);
+        var orderExists = await _orderRepository.OrderExistsAsync(reservationId, orderId);
+
+        if (!reservationExists)
+        {
+            return NotFound(ApiErrors.ReservationNotFound);
+        }
+
+        if (!orderExists)
+        {
+            return NotFound(ApiErrors.OrderNotFound);
+        }
+
+        var orderItem = await _orderItemRepository.GetOrderItemAsync(reservationId, orderId, orderItemId);
+        if (orderItem == null)
+        {
+            return NotFound(ApiErrors.OrderItemNotFound);
+        }
+
+        return Ok(_mapper.Map<OrderItemDto>(orderItem));
+    }
+
+    /// <summary>
+    /// Creates a new order item for a specific order in a reservation.
+    /// </summary>
+    /// <param name="reservationId">The ID of the reservation.</param>
+    /// <param name="orderId">The ID of the order.</param>
+    /// <param name="orderItemForCreation">The data for creating a new order item.</param>
+    /// <returns>The created order item.</returns>
+    /// <response code="201">Returns the created order item.</response>
+    /// <response code="404">If the reservation or order with the specified IDs is not found or Reservation is not consistent with MenuItem.</response>
+    /// <response code="400">If data is invalid</response>
+    [HttpPost]
+    [ProducesResponseType(StatusCodes.Status201Created, Type = typeof(OrderItemDto))]
+    [ProducesResponseType(StatusCodes.Status404NotFound)]
+    [ProducesResponseType(StatusCodes.Status400BadRequest)]
+    public async Task<ActionResult<OrderItemDto>> PostOrderItem(int reservationId, int orderId, OrderItemCreationDto orderItemForCreation)
+    {
+        var reservationExists = await _reservationRepository.ReservationExistsAsync(reservationId);
```

Comment:
when you use an additional service layer, you'll see the benefits, and the controller will look much cleaner